hat/assets/js/apps/Iaso/domains/forms/hooks/useGetPossibleFields.ts

```diff
@@ -41,6 +53,40 @@ export const useGetPossibleFields = (formId?: number): Result => {
     return usePossibleFields(isFetchingForm, currentForm);
 };
 
+export const useAllPossibleFields = (
+    isFetchingForms: boolean,
+    allForms: Form[] = [],
+): AllResults => {
+    return useMemo(() => {
+        let allPossibleFields: PossibleFieldsForForm[] = [];
+        allForms.forEach(form => {
+            const possibleFields =
+                form?.possible_fields?.map(field => ({
+                    ...field,
+                    fieldKey: field.name.replace('.', ''),
+                })) || [];
+            allPossibleFields.push({
+                form_id: form.form_id,
+                name: form.name,
+                possibleFields,
+            });
+        });
+
+        return {
+            allPossibleFields,
+            isFetchingForms,
+        };
+    }, [isFetchingForms]);
```

Comment:
```suggestion
    }, [isFetchingForms, allForms]);
```
You need to update the return value if `allForms` change. Normally your linter should warn you if your deps array is incomplete.

Reply:
If done on purpose, please leave a comment explaining the reason why, and why it's safe to do so

Reply:
Hm good catch, my linter didn't tell me anything about this, should it come from the `show-lint-problems` task?

Reply:
It's from `react-hooks/exhaustive-deps`